snuba/subscriptions/data.py

```diff
@@ -0,0 +1,46 @@
+from dataclasses import dataclass
+from datetime import datetime, timedelta
+from typing import List, Tuple
+
+from snuba.datasets.dataset import Dataset
+from snuba.datasets.factory import get_dataset
+from snuba.views import validate_request_content
+from snuba.query.types import Condition
+from snuba.request import Request
+from snuba.request.schema import RequestSchema
+
+
+@dataclass(frozen=True)
+class Subscription:
+    """
+    Represents the state of a subscription.
+    """
+    id: str
+    project_id: int
+    dataset: str
+    conditions: List[Condition]
+    aggregations: List[Tuple[str, str, str]]
+    time_window: int
+    resolution: int
```

Comment:
It might make sense to make these `timedelta` instances so we never have to have the "what unit is this again" conversation.